client/src/controllers/SessionController.ts

```diff
@@ -87,14 +87,6 @@ export class SessionController extends Controller<HTMLElement> {
   }
 
   connect(): void {
-    this.interceptTargets.forEach((button) => {
-      // Match the event listener configuration of workflow-action that uses
-      // capture so we can intercept the workflow-action's listener.
-      button.addEventListener('click', this.showConfirmationDialog, {
-        capture: true,
-      });
-    });
-
```

Comment:
The problem was, this controller is connected before the schedule publishing dialog is teleported to the edit form. As a result, `querySelectorAll` won't find the submit button yet.

We're fixing this by moving the query to `wDialogOutletConnected`, which is fired when the overwrite changes dialog template gets teleported and the its DialogController is connected. Fortunately, this happens after the schedule publishing dialog template is teleported, so its submit button is included in the query result.

This fix won't work for any submit button added to the editor form after that. We can't reliably detect the addition of those buttons without using MutationObserver on the form element and looking for new submit buttons, but I feel like that's an unnecessary overhead for something that we ourselves don't need.

Custom buttons that are added too late can still be hooked into the mechanism by adding the click event listener with the `SessionController.showConfirmationDialog` as the handler, so I think it's fair to leave it up to them to do this.